test/common/http/http1/conn_pool_test.cc

```diff
@@ -271,6 +272,53 @@ TEST_F(Http1ConnPoolImplTest, VerifyBufferLimits) {
   dispatcher_.clearDeferredDeleteList();
 }
 
+/**
+ * Verify that canceling pending connections within the callback works.
+ */
+TEST_F(Http1ConnPoolImplTest, VerifyCancelInCallback) {
+  Http::ConnectionPool::Cancellable* handle1{};
+  Http::ConnectionPool::Cancellable* handle2{};
+  // In this scenario, all connections must succeed, so when
+  // one fails, the others are canceled.
+  ConnPoolCallbacks callbacks1;
+  uint32_t pool_failure_calls{};
+  EXPECT_CALL(callbacks1.pool_failure_, ready())
+      .Times(AtMost(1))
```

Comment:
You should be able to use WillOnce here and below? This should be deterministic?

Reply:
essentially one and only one of those of those should be called (as one of them cancels the other). 
it depends on the implementation of the conn pool which one is first; while i can check the impl to see which one it will be, i thought it is better that the test covers the accepted behaviour to allow the impl to change without breaking the test.

Reply:
Point taken, but my preference would be simplify the test for now. Feel free to leave a comment about how it's tied to the implementation.